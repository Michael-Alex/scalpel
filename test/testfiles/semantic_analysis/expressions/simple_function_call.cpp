int f(int, double, unsigned int)
{
	return 0;
}

int i = 0;
int j = f(j, 1.8, 1u);
