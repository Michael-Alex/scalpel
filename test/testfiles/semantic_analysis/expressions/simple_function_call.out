{
  "id": "namespace-0",
  "simple functions":
  [
    {
      "id": "simple-function-0",
      "name": "f",
      "return type":
      {
        "fundamental type": "int"
      },
      "parameters":
      [
        {
          "type":
          {
            "fundamental type": "int"
          }
        },
        {
          "type":
          {
            "fundamental type": "double"
          }
        },
        {
          "type":
          {
            "fundamental type": "unsigned int"
          }
        }
      ]
    }
  ],
  "variables":
  [
    {
      "id": "variable-0",
      "name": "i",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "int": 0
      }
    },
    {
      "id": "variable-1",
      "name": "j",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "simple function call":
        {
          "function id": "simple-function-0",
          "arguments":
          [
            "variable": "variable-1",
            "double": 1.8,
            "unsigned int": 1
          ]
        }
      }
    }
  ]
}
